### ISSUE TITLE
Support online file access

### ISSUE THREAD

**Issue by pdawyndt** [dodona-edu/papyros#88]
This code should output a random haiku (scraped from a web page), but access to external URLS does not seem to be working.

```python
from urllib.request import urlopen

# download web page with random haiku
url = 'http://haikuguy.com/issa/random.php'
reader = urlopen(url)

# parse page until start of haiku is found
marker = '<p class="english">'
line = reader.readline().decode('utf-8')
while line and not line.startswith(marker):
    line = reader.readline().decode('utf-8')

# read three haiku lines and display them
if line.startswith(marker):
    print(line[len(marker):].strip()[:-6])
    line = reader.readline().decode('utf-8')
    print(line.strip()[:-6])
    line = reader.readline().decode('utf-8')
    print(line.strip()[:-4])
```

**Comment by alexmojaki** [comment]
See https://github.com/pyodide/pyodide/issues/662 and https://github.com/pyodide/pyodide/issues/375. You can read a URL with `pyodide.open_url` or `js.fetch`. I don't know why `urllib` isn't patched or something in pyodide.

In any case, the browser has many security restrictions which cause trouble for most URLs. Your example isn't `https` so it gets blocked. Other URLs are typically blocked by CORS.

**Comment by winniederidder** [comment]
@alexmojaki Is providing an override of urllib something you are doing in your own packages? Or should I make a Papyros-specific part? It would be similar to input/matplotlib by making urllib functions point to e.g. pyodide.open_url calls. The most common calls can easily be covered by this.

**Comment by alexmojaki** [comment]
I have no plans for this, go ahead.

**Comment by winniederidder** [comment]
After some studying, this issue would affect atleast the following libraries: `urllib`, `urllib3`, `http.client`, `requests`, SSL, `websocket-client`, `websockets`, and more.  Doing this in a maintainable way from our side is thus not really feasible. This is something that has to be fixed upstream and is thus a limitation of Pyodide. As @bmesuere mentioned, projects integrating Papyros will have protection mechanisms limiting the accessible URLs.
I will open a new issue to discuss limitations of Pyodide for things like this.

**Comment by bmesuere** [comment]
upstream issue
